storage/mysql/map_storage.go

```diff
@@ -106,32 +106,41 @@ func (m *mySQLMapStorage) begin(ctx context.Context, tree *trillian.Tree) (stora
 	}
 
 	mtx := &mapTreeTX{
-		treeTX: ttx,
-		ms:     m,
+		treeTX:       ttx,
+		ms:           m,
+		readRevision: -1,
 	}
 
-	mtx.root, err = mtx.LatestSignedMapRoot(ctx)
+	if readonly {
+		// readRevision will be set later, by the first
+		// GetSignedMapRoot/LatestSignedMapRoot operation.
+		return mtx, nil
+	}
+
+	root, err := mtx.LatestSignedMapRoot(ctx)
```

Comment:
Should the rest of this function be broken into a separate function that `GetSignedMapRoot` can also call?

Reply:
I don't think so -- both versions of `GetSignedMapRoot` (mapserver and storage) deal with `trillian.SignedMapRoot` objects rather than `types.MapRootV1` objects